app/controllers/application_controller.rb

```diff
@@ -132,6 +113,14 @@ def has_multiple_providers?
 
 private
 
+  def http_basic_auth
+    if Settings.basic_auth
+      authenticate_or_request_with_http_basic do |name, password|
+        name == "admin" && Digest::SHA512.hexdigest(password) == "52785638ec464fd61f5c9b372797f1a7475225cabeb2b40b2d757eff9b337ff069b2314bb0c0611d44ca5d39c91906ab3415de0fbc36625b970e3c2c03d122da"
```

Comment:
can we move 

`admin` and that `really long sha` into setting please.

Reply:
Done